react/components/SKUSelector/components/SKUSelector.js

```diff
@@ -1,69 +1,159 @@
 import PropTypes from 'prop-types'
-import React, { useCallback, memo } from 'react'
+import React, { useCallback, memo, useState, useEffect } from 'react'
 
 import Variation from './Variation'
-import { variationShape } from '../utils/proptypes'
+
+import { compose, flip, gt, filter, path, clone } from 'ramda'
 
 import styles from '../styles.css'
+import { variationShape, skuShape } from '../utils/proptypes'
+import { findItemWithSelectedVariations, findListItemsWithSelectedVariations, isColor } from '../utils'
+
+const isSkuAvailable = compose(flip(gt)(0), path(['sellers', '0', 'commertialOffer', 'AvailableQuantity']))
+
+const showItemAsAvailable = (possibleItems, selectedVariations, variationCount, isSelected) => {
+  const selectedNotNull = filter(Boolean, selectedVariations)
+  const selectedCount = Object.keys(selectedNotNull).length
+  if (selectedCount === variationCount && isSelected) {
+    const item = findItemWithSelectedVariations(possibleItems, selectedVariations)
+    return isSkuAvailable(item)
+  }
+  return possibleItems.some(isSkuAvailable)
+}
+
+const getAvailableVariations = ({ variations, selectedVariations, imagesMap, onSelectItemMemo, skuItems, hideImpossibleCombinations }) => {
```

Comment:
:pick:  why `onSelectItemMemo` instead of `onSelectItem`?

Reply:
what is `imagesMap`?

Reply:
I see now that `imagesMap` is commented below, on the propTypes. perhaps add a comment that these arguments are explained below? just a suggestion though